astropy/coordinates/representation.py

```diff
@@ -1193,16 +1489,17 @@ def scale_factors(self, omit_coslat=False):
 
     def represent_as(self, other_class):
         # Take a short cut if the other class is a spherical representation
-        if issubclass(other_class, PhysicsSphericalRepresentation):
-            return other_class(phi=self.lon, theta=90 * u.deg - self.lat,
-                               r=self.distance, copy=False)
-        elif issubclass(other_class, UnitSphericalRepresentation):
-            return other_class(lon=self.lon, lat=self.lat, copy=False)
-        else:
-            return super(SphericalRepresentation,
-                         self).represent_as(other_class)
+        if inspect.isclass(other_class):
```

Comment:
As @mhvk noted above for UnitSphericalRepresentation, these shortcuts drop the differentials